src/ui/render.ts

```diff
@@ -278,20 +293,31 @@ export default class UI {
 
   private registerEvents() {
     this.lightSheet.events.on(EventType.CORE_SET_CELL, (event) => {
-      if (this.lightSheet.isReady) this.onCoreSetCell(event);
+      this.onCoreSetCell(event);
     });
   }
 
   private onCoreSetCell(event: LightsheetEvent) {
     const payload = event.payload as CoreSetCellPayload;
-    // Get HTML elements and (new) IDs for the payload's cell and row.
-    const elInfo = LightSheetHelper.getElementInfoForSetCell(payload);
+    // Create new columns if the column index is greater than the current column count.
+    const columnCount = this.getColumnCount();
+    if (payload.indexPosition.column > columnCount - 1) {
+      const diff = payload.indexPosition.column + 1 - columnCount;
+      for (let i = 0; i < diff; i++) {
+        this.addColumn();
+      }
+    }
 
-    if (!elInfo.rowDom) {
-      const row = this.addRow(payload.indexPosition.row);
-      elInfo.rowDom = row;
-      row.id = elInfo.rowDomId;
+    const rowCount = this.getRowCount();
+    if (payload.indexPosition.row > rowCount - 1) {
+      const diff = payload.indexPosition.row + 1 - rowCount;
+      for (let i = 0; i < diff; i++) {
+        this.addRow();
+      }
     }
```

Comment:
The for-loop is never entered if i < 0, so this could be simplified to
```ts
const newColumns = payload.indexPosition.column - this.getColumnCount() + 1;
for (let i = 0; i < newColumns; i++) {
  this.addColumn();
}

const newRows = payload.indexPosition.row - this.getRowCount() + 1;
for (let i = 0; i < newRows; i++) {
  this.addRow();
}

Reply:
Very nice and elegant. Changed.